include/pmacc/Environment.hpp

```diff
@@ -449,26 +449,29 @@ namespace detail
         {
             throw std::runtime_error("no CUDA capable devices detected");
         }
-        else if (deviceNumber >= num_gpus) //check if device can be selected by deviceNumber
-        {
-            std::cerr << "no CUDA device " << deviceNumber << ", only " << num_gpus << " devices found" << std::endl;
-            throw std::runtime_error("CUDA capable devices can't be selected");
-        }
 #endif
 
         int maxTries = num_gpus;
-#if (PMACC_CUDA_ENABLED == 1)
-        cudaDeviceProp devProp;
-        CUDA_CHECK((cuplaError_t)cudaGetDeviceProperties(&devProp, deviceNumber));
-        /* if the gpu compute mode is set to default we use the given `deviceNumber` */
-        if (devProp.computeMode == cudaComputeModeDefault)
-            maxTries = 1;
-#endif
+        bool deviceSelectionSuccessfull = false;
+
         cudaError rc;
 
         for (int deviceOffset = 0; deviceOffset < maxTries; ++deviceOffset)
         {
             const int tryDeviceId = (deviceOffset + deviceNumber) % num_gpus;
+
+            log<ggLog::CUDA_RT>("Try to allocate device %1%.") % tryDeviceId;
+#if (PMACC_CUDA_ENABLED == 1)
+            cudaDeviceProp devProp;
+            CUDA_CHECK((cuplaError_t)cudaGetDeviceProperties(&devProp, tryDeviceId));
+            /* if the gpu compute mode is set to default we try to get a device only once */
```

Comment:
Not sure, but maybe this needs additional explanation. Like for now, it is not obvious for me that we do not break smth else with this change.

Reply:
I will extend the doc string a little bit and link the [cuda documentation](https://docs.nvidia.com/cuda/cuda-c-programming-guide/#compute-modes) for compute modes.